packages/vue-components/src/Modal.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <!-- NavBar has z-index of 1000, hence the z-index here should exceed that -->
+  <vue-final-modal
+    v-if="isMounted"
+    v-model="show"
+    ssr
+    :name="id"
+    classes="allow-overflow"
+    :content-class="['modal-dialog', optionalModalSize, optionalCentering]"
+    overlay-transition="none"
+    :transition="effectClass"
+    :click-to-close="backdrop !== 'false'"
+    esc-to-close
+    z-index-base="2000"
+  >
+    <div class="modal-content">
+      <div v-if="hasHeader" class="modal-header">
+        <h5 class="modal-title">
+          <slot name="header"></slot>
+        </h5>
+        <button
+          type="button"
+          class="close"
+          aria-label="Close"
+          @click="close()"
+        >
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <slot></slot>
+      </div>
+      <div v-if="hasFooter || hasOk" class="modal-footer">
+        <slot v-if="hasFooter" name="footer"></slot>
+        <button
+          v-else
+          class="btn btn-primary"
+          @click="close()"
+        >
+          {{ okText }}
+        </button>
+      </div>
+    </div>
+  </vue-final-modal>
+</template>
+
+<script>
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { VueFinalModal } from 'vue-final-modal';
+
+export default {
+  name: 'Modal',
+  components: {
+    VueFinalModal,
+  },
+  data: () => ({
+    show: false,
+    isMounted: false,
+    zoomEffect: {
+      'enter-class': 'modal-zoom',
+      'enter-to-class': 'modal-zoom-show',
+      'leave-to-class': 'modal-zoom',
+      'leave-class': 'modal-zoom-show',
+    },
+  }),
+  props: {
+    okText: {
+      type: String,
+      default: '',
+    },
+    effect: {
+      type: String,
+      default: 'zoom',
+    },
+    id: {
+      type: String,
+      default: '',
+    },
+    small: {
+      type: Boolean,
+      default: false,
+    },
+    large: {
+      type: Boolean,
+      default: false,
+    },
+    center: {
+      type: Boolean,
+      default: false,
+    },
+    backdrop: {
+      type: String,
+      default: '',
+    },
+  },
+  computed: {
+    hasHeader() {
+      return !!this.$slots.header;
+    },
+    hasFooter() {
+      return !!this.$slots.footer;
+    },
+    hasOk() {
+      return this.okText !== '';
+    },
+    optionalModalSize() {
+      if (!this.small && !this.large) {
+        return '';
+      }
+      return this.small ? 'modal-sm' : 'modal-lg';
+    },
+    optionalCentering() {
+      return this.center ? 'modal-dialog-centered' : '';
+    },
+    effectClass() {
+      return this.effect === 'zoom' ? this.zoomEffect : 'vfm';
+    },
+  },
+  methods: {
+    close() {
+      this.show = false;
+    },
+  },
+  mounted() {
+    this.isMounted = true;
+  },
+};
+</script>
+<style scoped>
```

Comment:
In view of #1768, we may want to consider using global styles instead since scoping may not be necessary here as these classes only appear in this file :)